apis/k8ssandra/v1alpha1/k8ssandracluster_webhook.go

```diff
@@ -192,6 +174,71 @@ func (r *K8ssandraCluster) ValidateUpdate(old runtime.Object) (admission.Warning
 	return nil, nil
 }
 
+func validateUpdateNumTokens(
+	oldCassandra *CassandraClusterTemplate,
+	newCassandra *CassandraClusterTemplate,
+) error {
+	oldNumTokensPerDc, err := numTokensPerDc(oldCassandra)
+	if err != nil {
+		return err
+	}
+	newNumTokensPerDc, err := numTokensPerDc(newCassandra)
+	if err != nil {
+		return err
+	}
+
+	for dcName, newNumTokens := range newNumTokensPerDc {
+		oldNumTokens, oldExists := oldNumTokensPerDc[dcName]
+		if oldExists && oldNumTokens != newNumTokens {
+			return ErrNumTokens
```

Comment:
We could add more details and keep iterating to get an exhaustive list, but I'm not sure if it's worth it.